src/main/java/core/basesyntax/suppliers/FigureSupplier.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax.suppliers;
+
+import core.basesyntax.figures.Circle;
+import core.basesyntax.figures.Figure;
+import core.basesyntax.figures.IsoscelesTrapezoid;
+import core.basesyntax.figures.Rectangle;
+import core.basesyntax.figures.RightTriangle;
+import core.basesyntax.figures.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+
+    public Figure getRandomFigure() {
+        int indexOfFigure = new Random().nextInt(4);
+        switch (indexOfFigure) {
+            case 0:
+                return new Circle(new ColorSupplier().getRandomColor(), getRandomDouble());
```

Comment:
lets create a separate private method to get each random figure, this will make code more readable
```suggestion
            case 0:
                return getRandomCircle();
```